MoreLinq/SkipLastWhile.cs

```diff
@@ -0,0 +1,62 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2024 Andy Romero (armorynode). All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Removes elements from the end of a sequence as long as a specified condition is true.
+        /// </summary>
+        /// <typeparam name="T">Type of the source sequence</typeparam>
+        /// <param name="source">The source sequence.</param>
+        /// <param name="predicate">The predicate to use to remove items from the tail of the sequence.</param>
+        /// <returns>
+        /// An <see cref="IEnumerable{T}"/> containing the source sequence elements except for the bypassed ones at the end.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">The source sequence is null.</exception>
+        /// <exception cref="ArgumentNullException">The predicate is null.</exception>
+
+        public static IEnumerable<T> SkipLastWhile<T>(this IEnumerable<T> source, Func<T, bool> predicate)
+        {
+            if (source == null) throw new ArgumentNullException(nameof(source));
+            if (predicate == null) throw new ArgumentNullException(nameof(predicate));
+
+            return _(source, predicate);
+
+            static IEnumerable<T> _(IEnumerable<T> source, Func<T, bool> predicate)
+            {
+                var list = source.ToArray();
```

Comment:
My original implementation was actually done using a `Queue`, but running initial benchmarks it seemed like the queue resulted in a slightly slower speed and a larger allocation. After running more benchmarks with more cases (empty sequences, only the first thousand items returned, all items returned, etc.) it looks like you're right, the queue is the better choice. _(especially since it can avoid Gen1 allocations altogether in certain cases as shown below)_.

| Method                  | N     | Mean      | Error    | StdDev   | Gen0    | Gen1    | Allocated |
|-------------------------|-------|-----------|----------|----------|---------|---------|-----------|
| SkipLastWhileArray      | 1,000 | 39.09 μs  | 0.070 μs | 0.062 μs | 2.5024  | 0.2441  | 39.16 KB  |
| SkipLastWhileList       | 1,000 | 42.63 μs  | 0.115 μs | 0.108 μs | 2.5024  | 0.2441  | 39.2 KB   |
| SkipLastWhileQueue      | 1,000 | 67.99 μs  | 0.226 μs | 0.200 μs | 0.4883  | -       | 8.35 KB   |
| SkipLastWhileStack      | 1,000 | 67.31 μs  | 0.234 μs | 0.207 μs | 0.4883  | -       | 8.34 KB   |

[LINQPad script](https://share.linqpad.net/3nf87km5.linq)